### PR TITLE
Comment Form

### PR DESCRIPTION
**Pull request by eriyezab**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+/**
+ * Class representing a comment that t users create.
```
Comment (ankitaj224) [inline portfolio/src/main/java/com/google/sps/data/Comment.java]: nit: the

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * <p>Note: The private variables in this class are converted into JSON.
+ */
+public class Comment {
```
Comment (ankitaj224) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:22]: Do you need any getters for these variables?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -26,22 +27,48 @@
...
+    // Redirect back to the comments page.
+    response.sendRedirect("/comments.html");
+    System.out.println("Redirecting user.");
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (ankitaj224) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:112]: indentation is off here. Please verify.

File: portfolio/src/main/webapp/comments.js [hunk 4]
```diff
@@ -7,7 +19,7 @@ function getComments() {
     for(i = 0; i < comments.length; ++i) {
       const commentsList = document.getElementById('comments-list');
       let listNode = document.createElement("LI");
-      let textNode = document.createTextNode(comments[i]);
+      let textNode = document.createTextNode(comments[i].name + ' at ' + timestampToDate(comments[i].timestamp) + ': ' + comments[i].message);
```
Comment (ankitaj224) [inline portfolio/src/main/webapp/comments.js]: Please check the indentation

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -26,22 +27,48 @@
...
     // Send json to server
     response.setContentType("applications/json;");
     response.getWriter().println(json);
+    System.out.println("Sent json to client.");
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get input from the form.
+    String name = getParameter(request, "name", "Anonymous");
```
Comment (ankitaj224) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:68]: Avoid adding to the comments if both name and message are set to default values.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -26,22 +27,48 @@
...
+    Comment newComment = new Comment(name, message, System.currentTimeMillis());
+    comments.add(newComment);
+    System.out.println("Added comment to list.");
+
+    // Redirect back to the comments page.
+    response.sendRedirect("/comments.html");
+    System.out.println("Redirecting user.");
+  }
+
+  /**
```
Comment (ankitaj224) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:103]: Javadoc for what the method does.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -26,22 +27,48 @@
...
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name).strip();
+    if (value == null || value == "") {
```
Comment (ankitaj224) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:114]: Why cant null be the default? And avoid adding to the list if form values are null?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -26,22 +27,48 @@
...
     response.getWriter().println(json);
+    System.out.println("Sent json to client.");
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get input from the form.
+    String name = getParameter(request, "name", "Anonymous");
+    String message = getParameter(request, "message", "No comment.");
+    System.out.println("Retireved input from form.");
```
Comment (ankitaj224) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:71]: If the print statements are for debugging purpose, I would recommend adding a TODO to remove them later. Else, if you are using them logging purpose - use Logger.log instead.

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 2 above)
Comment (eriyezab) [reply portfolio/src/main/java/com/google/sps/data/Comment.java:22]: At the current moment I do not. Should I add some anyway?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 6 above)
Comment (eriyezab) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:103]: Retrieve the value of the parameter \<name\> from the HTTP request \<request\>.

Would this be a good explanation? I am not too sure how I should insert the parameter names of the function into the comment. Is the way I did it okay or is there a preferred way.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 7 above)
Comment (eriyezab) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:114]: Just added this functionality!

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 2 above)
Comment (ankitaj224) [reply portfolio/src/main/java/com/google/sps/data/Comment.java:22]: Add a TODO.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 6 above)
Comment (ankitaj224) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:103]: https://www.tutorialspoint.com/java/java_documentation.htm

Use @param for method params

**Review by ankitaj224 — APPROVED** [review]
Minor nits.

**Pull request merged**